compiler/src/dotty/tools/dotc/Driver.scala

```diff
@@ -30,18 +31,20 @@ class Driver {
 
   protected def doCompile(compiler: Compiler, files: List[AbstractFile])(using Context): Reporter =
     if files.nonEmpty then
+      var run1 = ctx.run
       try
         val run = compiler.newRun
+        run1 = run
         run.compile(files)
         finish(compiler, run)
       catch
         case ex: FatalError =>
-          report.error(ex.getMessage.nn) // signals that we should fail compilation.
-        case ex: TypeError =>
-          println(s"${ex.toMessage} while compiling ${files.map(_.path).mkString(", ")}")
+          report.error(run1.enrichErrorMessage(ex.getMessage.nn)) // signals that we should fail compilation.
```

Comment:
It's too late to capture the context here. Unless I'm mistaken, the tree, symbol, position, even source file are all going to be wrong at this point, because the context they existed in got blown away many stack frames ago. see my last para here https://github.com/lampepfl/dotty/pull/16593#issuecomment-1369469965 

@smarter's comment in the next reply seems to back this up

Reply:
Yeah, this case effectively handles when compiler.newRun fails.  If it succeeds, then there's the handling in `run.compile`.  I just applied the same strategy to these existing catch cases, to avoid it spamming, in particular I took out re-printing the exception message.